smartsim/_core/launcher/dragon/dragonBackend.py

```diff
@@ -63,10 +83,72 @@ class DragonBackend:
     by threads spawned by it.
     """
 
-    def __init__(self) -> None:
-        self._proc_groups: t.Dict[str, t.Tuple[ProcessGroup, t.List[int]]] = {}
+    def __init__(self, pid: int) -> None:
+        self._pid = pid
+        self._group_infos: t.Dict[str, ProcessGroupInfo] = {}
         self._step_id_lock = RLock()
+        self._hostlist_lock = RLock()
         self._step_id = 0
+        # hosts available for execution
+        # dictionary maps hostname to step_id of
+        # step being executed on it
+        self._initialize_hosts()
+        # Suppress pylint error which only happens for Python 3.7
+        # pylint: disable-next=unsubscriptable-object
+        self._queued_steps: collections.OrderedDict[str, DragonRunRequest] = (
+            collections.OrderedDict()
+        )
```

Comment:
if you want to remove the pylint suppression:
```suggestion
        self._queued_steps: "collections.OrderedDict[str, DragonRunRequest]" = (
            collections.OrderedDict()
        )
```

Reply:
Applied